src/Libraries/Nop.Services/Affiliates/AffiliateService.cs

```diff
@@ -74,7 +74,7 @@ public virtual async Task<Affiliate> GetAffiliateByFriendlyUrlNameAsync(string f
 
             var query = from a in _affiliateRepository.Table
                         orderby a.Id
-                        where a.FriendlyUrlName == friendlyUrlName
+                        where a.FriendlyUrlName.ToLower() == friendlyUrlName.ToLower()

```

Comment:
you should not mix changes in the source code and adding unit tests, this confuses and makes it impossible to determine where and why the source code was changed. Please add another pull request with these changes if you are sure that they need and describe why you think so